Sources/TuistGenerator/Graph/Graph.swift

```diff
@@ -221,10 +221,14 @@ class Graph: Graphing {
         references = references.union(precompiledLibrariesAndFrameworks)
 
         // Static libraries and frameworks
-
+        var staticLibraryTargetNodes = [TargetNode]()
         if targetNode.target.canLinkStaticProducts() {
-            let staticLibraries = findAll(targetNode: targetNode, test: isStaticLibrary, skip: isFramework)
-                .map { DependencyReference.product(target: $0.target.name) }
+            var staticLibraries = [DependencyReference]()
+
+            findAll(targetNode: targetNode, test: isStaticLibrary, skip: isFramework).forEach {
```

Comment:
we could keep the static libraries related code local to this scope via some minor tweaks:

e.g.

```swift
let staticLibraryTargetNodes = findAll(targetNode: targetNode, test: isStaticLibrary, skip: isFramework)
let staticLibraries = staticLibraryTargetNodes.map {
     DependencyReference.product(target: $0.target.name)
}

let staticDependenciesDynamicLibraries = staticLibraryTargetNodes.flatMap {
    $0.targetDependencies
           .filter(or(isFramework, isDynamicLibrary))
           .map { DependencyReference.product(target: $0.target.name) }
 }

references = references.union(staticLibraries + staticDependenciesDynamicLibraries)
...
```

I guess that sorta goes against the sections marked out in comments - though those could change too if that helps

Reply:
I had something similar before but changed it due to the comments in the code. I'll change the comments and go by your suggestion.